src/Apollo/NcrReactionsProjector.php

```diff
@@ -0,0 +1,216 @@
+            return;
+        }
+
+        $params = [
+            'TableName'                 => $tableName,
+            'Key'                       => [
+                NodeTable::HASH_KEY_NAME => ['S' => $reactionsRef->toString()],
+            ],
+            'UpdateExpression'          =>  rtrim('set'. $updateExpression, ', '),
+            'ExpressionAttributeNames'  => $expressionAttributeNames,
+            'ExpressionAttributeValues' => [
+                ':v_incr' => ['N' => '1'],
+            ],
+            'ReturnValues'              => 'NONE',
+        ];
+
+        $this->client->updateItem($params);
+        $this->ncrSearch->indexNodes([$this->ncr->getNode($reactionsRef, true, $context)], $context);
+    }
+
+    protected function projectNode(Message $reactions, Message $event, Pbjx $pbjx): void
+    {
+        $context = ['causator' => $event];
+        $reactions
+            ->set('updated_at', $event->get('occurred_at'))
+            ->set('updater_ref', $event->get('ctx_user_ref'))
+            ->set('last_event_ref', $event->generateMessageRef())
+            ->set('etag', Aggregate::generateEtag($reactions));
+
+        $this->ncr->putNode($reactions, null, $context);
+        $this->ncrSearch->indexNodes([$reactions], $context);
+        $pbjx->trigger($reactions, 'projected', new NodeProjectedEvent($reactions, $event), false, false);
+    }
+
+    protected function createReactions(NodeRef $nodeRef): Message
+    {
+        static $class = null;
+        if (null === $class) {
+            $class = MessageResolver::resolveCurie('*:apollo:node:reactions:v1');
+        }
+
+        $reactions = $class::fromArray(['_id' => $nodeRef->getId()]);
+        $this->addReactionTypes($reactions);
+
+        return $reactions;
+    }
+
+    protected function getReactions(NodeRef $nodeRef, Message $event): ?Message
+    {
+        try {
+            $reactionsRef = $this->createReactionsRef($nodeRef);
+            return $this->ncr->getNode($reactionsRef, true, ['causator' => $event]);
+        } catch (NodeNotFound $nf) {
+            // Node not found do nothing here
+        } catch (\Throwable $e) {
+            throw $e;
+        }
+
+        return null;
+    }
+
+    protected function addReactionTypes(Message $reactions): void
+    {
+        foreach ([
+                     'love',
+                     'haha',
+                     'wow',
+                     'wtf',
+                     'trash',
+                     'sad',
+                 ] as $reactionType) {
+            $reactions->addToMap('reactions', $reactionType, 0);
+        }
+    }
+
+    protected function mergeNode(Message $node, Message $reactions): void
+    {
+        $reactions
+            ->set('_id', $node->get('_id'))
+            ->set('status', $node->get('status'))
+            ->set('title', $node->get('title'))
+            ->set('target', $node::schema()->getQName()->getMessage());
+
+        if ($node->has('published_at')) {
```

Comment:
@bhsiao55 doesnt this mean that the created_at time will change? if it had been created before the other content was publisehd. i guess thats how stats are now but seems odd

Reply:
@efjacobson yes the created_at on the reactions node will change once the article is published.  The stats do it this way so I just did the same